File: template/src/main/java/ru/surfstudio/standard/ui/util/camera/CameraIntentHelperCallback.java
package ru.surfstudio.standard.ui.util.camera;

import android.net.Uri;

import java.util.Date;

/**
 * Specifies the interface of a CameraIntentHelper request. The calling class has to implement the
 * interface in order to be notified when the request completes, either successfully or with an error.
 *
 * @author Ralf Gehrer <ralf@ecotastic.de>
 */
public interface CameraIntentHelperCallback {
    void onPhotoUriFound(Date dateCameraIntentStarted, Uri photoUri, int rotateXDegrees);

    void deletePhotoWithUri(Uri photoUri);

    void onSdCardNotMounted();

    void onCanceled();

    void onCouldNotTakePhoto();

    void onPhotoUriNotFound();

    void logException(Exception e);
}

File: template/src/main/java/ru/surfstudio/standard/ui/util/camera/CameraIntentHelperCallback.kt
package ru.surfstudio.standard.ui.util.camera

import android.net.Uri

import java.util.Date

/**
 * Specifies the interface of a CameraIntentHelper request. The calling class has to implement the
 * interface in order to be notified when the request completes, either successfully or with an error.
 *
 * @author Ralf Gehrer <ralf></ralf>@ecotastic.de>
 */
interface CameraIntentHelperCallback {
    fun onPhotoUriFound(dateCameraIntentStarted: Date?, photoUri: Uri?, rotateXDegrees: Int)

    fun deletePhotoWithUri(photoUri: Uri)

    fun onSdCardNotMounted()

    fun onCanceled()

    fun onCouldNotTakePhoto()

    fun onPhotoUriNotFound()

    fun logException(e: Exception)
}